sky/cli.py

```diff
@@ -2928,6 +2934,9 @@ def spot_launch(
         disk_size=disk_size,
         spot_recovery=spot_recovery,
     )
+    controller_resources = spot_lib.DEFAULT_SPOT_CONTROLLER_RESOURCES
+    controller_resources = controller_resources.copy(
```

Comment:
add `if controller_type is not None:`?

Reply:
It should have similar behaviro, but yes, adding it may be easier to read.

Reply:
It should have similar behaviro, but yes, adding it may be easier to read. Added. Thanks! ; )